client/web/src/repo/commits/RepositoryCommitsPage.tsx

```diff
@@ -131,7 +142,7 @@ export const RepositoryCommitsPage: React.FunctionComponent<React.PropsWithChild
     useBreadcrumb(BREADCRUMB)
 
     const queryCommits = useCallback(
-        (args: FilteredConnectionQueryArguments): Observable<GQL.IGitCommitConnection> => {
+        (args: FilteredConnectionQueryArguments): Observable<RepositoryCommitConnection> => {
```

Comment:
I understand the need to extract GQL fields into fragments to reference them as Typescript types, but it creates hard-to-follow GraphQL queries when multiple fragments are nested. We can use Typescript utility types to avoid deep nesting of GQL fragments.

```suggestion
        (args: FilteredConnectionQueryArguments): Observable<NonNullable<RepositoryGitCommitsRepository['commit']>['ancestors']> => {
```

Let's remove one level of extra nesting using this technique. Please apply the same approach to other queries updated in this PR.